src/Application.js

```diff
@@ -129,9 +128,11 @@ export class Application {
 
     Application._plugins.forEach(plugin => plugin.setup(this));
 
-    const preloads = Application._plugins.map(plugin => plugin.preload(this));
+    const preloader = Application._plugins.reduce((promise, plugin) => {
```

Comment:
I think I understand what's going on here but it's kind of confusing. Might be worth adding some comments just to clarify what's going on.

Reply:
Updated. Let me know if that reads a little better